pilot/pkg/model/push_context.go

```diff
@@ -320,11 +326,41 @@ func (ps *PushContext) initVirtualServices(env *Environment) error {
 		return err
 	}
 
+	type combinedVirtualService struct {
+		// What are the parameters to order?
+		// Virtual services have multiple hosts and multiple protocols. We need to split it into
+		// a multiple virtual services each with one host and one protocol type http/tcp/tls.
+		// then we need to look at the set of all virtual services for the same host/protocol
+		// and try to merge these services into one
+		//   Whats the ordering algorithm?
+		//    For http, sort all the match conditions for the virtual services for the host
+		//     specific port before no port
+		//     specific gateway before no gateway
+		//     specific source label before no source label
+		//     uri match before no uri match
+		//         between two URIs,
+		//           exact before prefix
+		//           prefix before regex
+		//           longer value (exact/prefix/regex) before the shorter one
+		//     authority match before no authority match
```

Comment:
authority is part of the URI so I suggest that we do this before path and change the verbiage above to refer to path